### PR TITLE
Add release notes

### PR DESCRIPTION
**Pull request by rollandf**
Source:
https://docs.nvidia.com/networking/display/kubernetes2410/release+notes

### CODE REVIEW

File: docs/release-notes.rst [hunk 1]
```diff
@@ -19,4 +19,179 @@
...
+       | - Added support for Network Operator upgrade from v1.0.0.
+       | - Added support for Kubernetes POD Security Policy.
+       | - Added support for Kubernetes >= 1.17 and <=1.22.
+       | - Added the ability to propagate nodeAffinity property from the NicClusterPolicy to Network Operator dependencies.
+   * - 1.0.0
+     - | - Added Node Feature Discovery that can be used to mark nodes with NVIDIA SR-IOV NICs.
+       | - Added support for different networking models:
+       |     - Macvlan Network
+       |     - HostDevice Network
+       |     - SR-IOV Network
```
Comment (vasrem) [inline docs/release-notes.rst:101]: (nothing to address) Interesting, this nesting doesn't seem to work on github preview but works on the output html

File: docs/release-notes.rst [hunk 2]
```diff
@@ -19,4 +19,179 @@
...
+   :local:
+   :backlinks: none
+
+========================
+Changes and New Features
+========================
+
+.. list-table::
+   :header-rows: 1
+   :widths: 10, 75
```
Comment (vasrem) [inline docs/release-notes.rst:35]: Do we need the widths? I guess it can auto adjust based on the text

**Review by vasrem — APPROVED** [review]
LGTM

Do we need those sections too?
* https://docs.nvidia.com/networking/display/kubernetes2310/release+notes#src-144713493_ReleaseNotes-System_Requrirements
* https://docs.nvidia.com/networking/display/kubernetes2310/release+notes#src-144713493_ReleaseNotes-TestedNetworkAdapters
* https://docs.nvidia.com/networking/display/kubernetes2310/release+notes#src-144713493_ReleaseNotes-Prerequisites
* https://docs.nvidia.com/networking/display/kubernetes2310/release+notes#src-144713493_ReleaseNotes-Versions

File: docs/release-notes.rst (hunk 2 above)
Comment (rollandf) [reply docs/release-notes.rst:35]: Here how it looks without width
image

**Comment by rollandf** [comment]
> LGTM
> 
> Do we need those sections too?
> 
> * https://docs.nvidia.com/networking/display/kubernetes2310/release+notes#src-144713493_ReleaseNotes-System_Requrirements
> * https://docs.nvidia.com/networking/display/kubernetes2310/release+notes#src-144713493_ReleaseNotes-TestedNetworkAdapters
> * https://docs.nvidia.com/networking/display/kubernetes2310/release+notes#src-144713493_ReleaseNotes-Prerequisites
> * https://docs.nvidia.com/networking/display/kubernetes2310/release+notes#src-144713493_ReleaseNotes-Versions

I am looking at 24.1 version

**Pull request merged**